models/foreignreference.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 Gitea. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/models/db"
+)
+
+// ForeignReference represents external references
+type ForeignReference struct {
+	LocalID   int64  `xorm:"INDEX"`
+	ForeignID string `xorm:"INDEX UNIQUE(external_reference_index)"`
```

Comment:
The `ForeignID` is string in database, but in code, it's `int64`, I think the type should be consistent.